Include/object.h

```diff
@@ -123,10 +123,10 @@ typedef struct {
 #define Py_TYPE(ob)             (_PyObject_CAST(ob)->ob_type)
 #define Py_SIZE(ob)             (_PyVarObject_CAST(ob)->ob_size)
 
-static inline int _Py_IS_TYPE(PyObject *ob, PyTypeObject *type) {
+static inline int _Py_IS_TYPE(const PyObject *ob, const PyTypeObject *type) {
     return ob->ob_type == type;
 }
-#define Py_IS_TYPE(ob, type) _Py_IS_TYPE(_PyObject_CAST(ob), type)
```

Comment:
I would prefer to have a new private _PyObject_CAST_CONST() macro rather than casting directly.

I prefer macros because maybe tomorrow we might be able to check if the pointer is valid PyObject in debug mode. Maybe using _PyObject_CheckConsistency().

Would you mind to move these changes (add the macro and modify Py_IS_TYPE to use const) into a separated PR?

Reply:
Sure, I'll make two separate PRs.

Reply:
They are now in #18798 and #18799